cyclops-ui/src/components/pages/ModuleDetails.tsx

```diff
@@ -144,6 +152,40 @@ const ModuleDetails = () => {
 
   let { moduleName } = useParams();
 
+  const [showManagedFields, setShowManagedFields] = useState(false);
+
+
+  const handleCheckboxChange = (e: CheckboxChangeEvent) => {
+    setShowManagedFields(e.target.checked);
```

Comment:
You can call the fetchManifest here directly instead of using `useEffect` on line 176. Also, you have some errors because of useEffect dependencies so you can delete it and just call `fetchManifest` here

Reply:
@petar-cvit if I call fetchManifest  after the setShowManagedFields on line 160, it's not working as we wanted because managedFields is not updating immediately  it is working in reverse order, means if checkbox is checked then managedFields are not displayed and when we unchecked the checkbox then managedFields are displayed. I have tried from  my end but unable to solve this without useEffect, could you please guide me.

Reply:
@Wasiim105
So I was thinking of fetching the manifest and passing it `e.target.checked` (you will also need to add that argument to the `fetchManifest` function).

```typescript
const handleCheckboxChange = (e: CheckboxChangeEvent) => {
  setShowManagedFields(e.target.checked);
  fetchManifest(
    manifestModal.resource.group,
    manifestModal.resource.version,
    manifestModal.resource.kind,
    manifestModal.resource.namespace,
    manifestModal.resource.name,
    e.target.checked,
  );
};
```

You can store it to the state (`setShowManagedFields`) so you can use it in the `handleManifestClick`. In that function you can add the fetchManifest and pass whatever was in the `showManagedFields` state

```typescript
  const handleManifestClick = (resource:any) => {
    setManifestModal({
      on: true,
      resource: {
        group: resource.group,
        version: resource.version,
        kind: resource.kind,
        name: resource.name,
        namespace: resource.namespace
      },
      manifest: ''
    });
    fetchManifest(
        resource.group,
        resource.version,
        resource.kind,
        resource.namespace,
        resource.name,
        showManagedFields,
    );
  };
```

With these changes you can delete the useEffect which requires you add a lot of dependencies which was the main reason why I am rooting for removing it. Let me know what you think

Reply:
Thank you for helping me. Now I know where I made a mistake.